### PR TITLE
Add Sort and Undo command to UG

### LINKED ISSUE #182: Update UG for Sort-by and Undo command

### PR DESCRIPTION
**Pull request by Echomo-Xinyu**
Fixes #182

### CODE REVIEW

File: docs/UserGuide.md [hunk 1]
```diff
@@ -260,13 +262,34 @@ Example:
...
 
 ---
 
+### Sort the displayed students: 'sort'
+
+Sort the displayed list of students (in the Student's Details panel) by the given `type` and `order`.
+
+Format: `sort TYPE [ORDER]`
+
+- `TYPE` must be from `NAME`, `CLASS`, and `OWED`.
```
Comment (sjoann) [inline docs/UserGuide.md:278]: will be better if presented as
`TYPE` must be followed by either one of the below options.
- `NAME`
- `CLASS`
- `OWED`
similar for the ORDER as well

File: docs/UserGuide.md [hunk 2]
```diff
@@ -260,13 +262,34 @@ Example:
...
+- `ORDER`, if present, must be from `ASC` or `DESC`. If it is left blank, it will follow the default order for the type.
+- Default order for `NAME` and `CLASS` is `ASC` while default order for `OWED` is `DESC`.
+- The two parameters `TYPE` and `ORDER` can be non-capitalized such as `name`, `nAMe` and so on.
+
+Examples:
+
+- `list` followed by `sort NAME` will list all the students by ascending order of their names.
```
Comment (sjoann) [inline docs/UserGuide.md:285]: Think can word as:
`list` followed by `sort NAME` will list all the students according to their names in ascending order.
similar for below

File: docs/UserGuide.md [hunk 3]
```diff
@@ -282,7 +305,7 @@ Examples:
 - `find Betsy` followed by `delete 1` deletes the 1st person in the Student's Details panel.
 
 ```yaml
-❗ Caution: Deleting a student is irreversible! Please input the correct index number(s).
+❗ Caution: Deleting a student by mistake can be reversed by "undo" command.
```
Comment (sjoann) [inline docs/UserGuide.md:308]: not sure if can add a hyperlink here inside yaml. if can, would be good :)

File: docs/UserGuide.md [hunk 4]
```diff
@@ -296,13 +319,23 @@ Clears all students and their details from the list.
 Format: `clear`
 
 ```yaml
-❗ Caution: Clearing all students is irreversible!
+❗ Caution: Clearing all students by mistake can be reversed by "undo" command!
```
Comment (sjoann) [inline docs/UserGuide.md:322]: similar as above

File: docs/UserGuide.md [hunk 5]
```diff
@@ -15,8 +15,10 @@ Teacher’s Pet can get your contact and class management tasks done faster than
     * [Editing student details: `edit`](#editing-student-details-edit)
     * [Viewing all students: `list`](#viewing-all-students-list)
     * [Finding a student: `find`](#finding-a-student-find)
+    * [Sort displayed students: `sort`](#sort-the-displayed-students-sort)
```
Comment (sjoann) [inline docs/UserGuide.md:18]: would be good to add nested hyperlinks for 
- Sort by name
- Sort by class date
- Sort by money owed

**Review by sjoann — APPROVED** [review]
LGTM! just some minor cosmetic issues and good to include more hyperlinks :)

File: docs/UserGuide.md (hunk 5 above)
Comment (Echomo-Xinyu) [reply docs/UserGuide.md:18]: The three fields are not broken to go under different subheadings as their contents are very similar. There would not be a hyperlink for these sub-points tho.

File: docs/UserGuide.md (hunk 3 above)
Comment (Echomo-Xinyu) [reply docs/UserGuide.md:308]: Have researched on this topic and cannot seem to find a quick fix :(

File: docs/UserGuide.md (hunk 5 above)
Comment (Echomo-Xinyu) [reply docs/UserGuide.md:31]: I have figured out what do you mean by this comment after examining the recent changes and will follow in the upcoming commit.

**Pull request merged**